File: 2825-Make String a Subsequence Using Cyclic Increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        idx2 = 0
        N = len(str2)
        for i in range(len(str1)):
            if str2[idx2] == str1[i] or chr( (ord(str1[i]) - ord('a') + 1)%26 + ord('a')) == str2[idx2]:
                idx2 += 1
            if idx2 == N:
                return True

        return False

File: C++/2825-Make String a Subsequence Using Cyclic Increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        int idx2 = 0;
        for (char c:str1) {
            if (c == str2[idx2] || ((c-'a'+1)%26 == str2[idx2]-'a')) {
                idx2++;
            }
            if (idx2 == str2.size()) return true;
        }
        return false;
    }
};